rust/src/database/database_manager.rs

```diff
@@ -149,8 +155,18 @@ impl DatabaseManager {
 
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
     pub(crate) async fn get_cached_or_fetch(&self, name: &str) -> Result<Arc<Database>> {
-        let cached_database = self.databases_cache.read().unwrap().get(name).cloned();
-        Ok(cached_database.unwrap_or(self.get(name).await?))
```

Comment:
`unwrap_or` resolves the value inside of it, that's the place where we spent time.